File: programmers/최소직사각형.py
"""
코딩테스트 연습 > 완전탐색
"""


def solution(sizes):
    sizeA = 0
    sizeB = 0

    for size in sizes:
        sizeA = max(sizeA, max(size[0], size[1]))
        sizeB = max(sizeB, min(size[0], size[1]))

    return sizeA * sizeB

File: programmers/최소직사각형.kt
import kotlin.math.*

class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var weight = 0
        var height = 0

        sizes.forEach {
            weight = max(weight, max(it[0], it[1]))
            height = max(height, min(it[0], it[1]))
        }

        return weight * height
    }
}
